src/Deserialization/ModuleFactory.cs

```diff
@@ -35,10 +35,10 @@ public IModule Create(LLVMModuleRef module, Bytes pointerSize)
                 CreateStructType(module, "struct.__va_list_tag"),
                 module.GetFunctions()
                     .Select(_functionFactory.Create)
-                    .ToArray(),
+                    .ToDictionary(f => f.Id),
```

Comment:
Smallish point but I wonder if having a static `Create` function on `Module` that accepts an `IEnumerable` and effectively internalises the `ToDictionary` call would be better encapsulation here as the `Dictionary` part seems like an implementation detail of `Module`.

Reply:
I'll do another PR for this very shortly if you don't mind. It conflicts with something else I have already done and also I will be changing this is maybe 4 places for consistency so would be better if you had a proper look at just that.